vscode/grammars/ruby.cson.json

```diff
@@ -100,6 +100,51 @@
       "match": "(?<!\\.)\\belse(\\s)+if\\b",
       "name": "invalid.deprecated.ruby"
     },
+    {
+      "captures": {
+        "1": {
+          "name": "variable.ruby"
+        },
+        "3": {
+          "name": "keyword.operator.assignment.augmented.ruby"
+        }
+      },
+      "match": "^\\s*([a-z]([A-Za-z0-9_])*)\\s*(&&=)",
+      "comment": "A local variable and assignment"
+    },
+    {
+      "captures": {
+        "1": {
+          "name": "variable.ruby"
+        },
+        "3": {
+          "name": "keyword.operator.assignment.augmented.ruby"
+        }
+      },
+      "match": "^\\s*([a-z]([A-Za-z0-9_])*)\\s*(\\|\\|=)",
+      "comment": "A local variable or assignment"
+    },
+    {
+      "captures": {
+        "1": {
+          "name": "variable.ruby"
+        },
+        "3": {
+          "name": "keyword.operator.assignment.augmented.ruby"
+        }
+      },
+      "match": "^\\s*([a-z]([A-Za-z0-9_])*)\\s*((\\+|\\*|-|\\/)=)",
+      "comment": "A local variable operation assignment (+=, -=, *=, /=)"
+    },
+    {
+      "captures": {
+        "1": {
+          "name": "variable.ruby"
+        }
+      },
+      "match": "^\\s*([a-z]([A-Za-z0-9_])*)\\s*=[^=]",
+      "comment": "A local variable assignment"
```

Comment:
Comparing against the list in https://github.com/ruby/irb/pull/961, I think there also are `%=`, `**=`, `&=`, `|=`, `^=`, `<<=`, `>>=`

Reply:
Ah, nice! Thanks for sharing the others. I added a new commit to improve the regex.